### ISSUE TITLE
error in first train gen loss=0.0 

### ISSUE THREAD

**Issue by lpscr** [yl4579/StyleTTS2#206]
hi thank you very much for your amazing work this so amazing and i want to train for the scratch non English
when i start the first  train ,  start to train, but all the epoch its get loss=0.0 dis loss =0.0, etc
only Mel Loss update , after i finish the 50 epoch i send image to see how look like 

![er1](https://github.com/yl4579/StyleTTS2/assets/147736764/3683270c-5c33-4b31-9431-d2eac25d0283)

i finish 50 epoch for 4 hours database for testing only

i start now the train second but i get error,  not start it give me error about the trace and stuck and to close the terminal or notebook
```
#in this line code
 running_loss += loss_mel.item()
            g_loss.backward()
            if torch.isnan(g_loss):
                from IPython.core.debugger import set_trace
                set_trace()

            optimizer.step('bert_encoder')
            optimizer.step('bert')
            optimizer.step('predictor')
            optimizer.step('predictor_encoder')
```
here screenshot how look like when it's stuck in train second       

![er2](https://github.com/yl4579/StyleTTS2/assets/147736764/9f53166c-478d-4abc-b600-733149e90b1b)

also i i try with the accelerate launch --mixed_precision=fp16 with and out --mixed_precision
and also with out accelerate  , simple runt with python and same problem

i use the default  config.yml 

batch to 20 
max len 300 
also i try change to change to other value the bath like 15 max len 100 same problem again

i try in rent gpu, A6000 or in my machine RTX4090 with wsl linux and same problem again 
i try also other database same again problem 
i use database about 5 hours single speaker   ,
 itry with 1,4,6 hours also

with format in database train,val i use about sample 5000 , and val 500
with this format ,
wav file | phonimize| 0

so what i do wrong here ?

**Comment by lpscr** [comment]
Today, I created a new training session, leaving it to run for about 60 epochs after 50th epoch when it began to update the gloss but when i run the train second stuck again :(  Could you please check the Tensor Board to see if this behavior is normal?
here the Tenterboard 

![image](https://github.com/yl4579/StyleTTS2/assets/147736764/aee7c7ff-5ff7-439b-a561-398ec1a15530)

**Comment by lpscr** [comment]
i try train again 30h dataset new complete take 3 days now for train first, i stop in epoch 75 and when i run the second train again same problem :(

i am not sure what i do wrong , i need more epochs ? to work this problem please someone if can tell me to because this take days to finish now i use RTX 4090 why i get this error in second train i dont understand

**Comment by yl4579** [comment]
I think it means your loss becomes NaN, can you print all the variables and see if any of them is NaN?

**Comment by lpscr** [comment]
@yl4579 hi thank you very much for the reply there is not NAN in loss only 0 after 50 epoch i get values here the train log 

[train.log](https://github.com/yl4579/StyleTTS2/files/14538714/train.log)

**Comment by lpscr** [comment]
here also tenser board 30h dataset i try 76 epochs i stop
![image](https://github.com/yl4579/StyleTTS2/assets/147736764/bd9db938-4ff7-4f8d-b761-cee14727a4ce)

**Comment by martinambrus** [comment]
issue #254 as well as its connected PR #253 solved this issue for me

**Comment by iAdityaVishnu** [comment]
@lpscr I just started training from scratch, but I am getting Nan in the first 2 epochs itself for training one. And i have made no changes in the code. Any idea what is wrong?
